CHANGES.rst

```diff
@@ -312,6 +312,9 @@ astropy.stats
 astropy.table
 ^^^^^^^^^^^^^
 
+- Fix a bug where slices of columns still carried indices even if they lost
```

Comment:
I think this should be called an API change. It was not wrong or unexpected behavior before given the original design in `BaseColumnInfo` (that indices were copied), but we are deciding now that it is more logical / better performance to change the design. Also, although it would be surprising, someone might be expecting the old behavior.

Reply:
OK to mention it as an API change. The reason I thought it might be more like a bug is that `parent_table` is not copied, which seems to make the indices rather useless. Furthermore, if the slice is taken on the table, the indices are not just copied, but carry information about the slice, so the behaviour is not quite consistent (see below). If it is a bug, in principle, we could backport this to 4.0.

```
In [63]: ts_a
Out[63]: 
<TimeSeries length=5>
          time            flux 
                          mJy  
         object         float64
----------------------- -------
2016-03-22T12:30:31.000     1.0
2016-03-22T12:30:34.000     4.0
2016-03-22T12:30:37.000     5.0
2016-03-22T12:30:40.000     3.0
2016-03-22T12:30:43.000     2.0

In [64]: ts_a['time'].info.indices
Out[64]: 
[          time          rows
 ----------------------- ----
 2016-03-22T12:30:31.000    0
 2016-03-22T12:30:34.000    1
 2016-03-22T12:30:37.000    2
 2016-03-22T12:30:40.000    3
 2016-03-22T12:30:43.000    4]

In [65]: ts_a[:3]['time'].info.indices
Out[65]: 
[Index slice (0, 3, 1) of
           time          rows
 ----------------------- ----
 2016-03-22T12:30:31.000    0
 2016-03-22T12:30:34.000    1
 2016-03-22T12:30:37.000    2
 2016-03-22T12:30:40.000    3
 2016-03-22T12:30:43.000    4]
```